client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java

```diff
@@ -56,50 +62,70 @@
 
     private final PlaceManager placeManager;
     private final TokenFormatter tokenFormatter;
-
-    @Override
-    protected void onAttach() {
-        PlaceRequest currentPlaceRequest = placeManager.getCurrentPlaceRequest();
-
-        setAnchor(currentPlaceRequest, NameTokens.GAE_STUDIO, gaeAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.ARCBEES, arcbeesAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.CHOSEN, chosenAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.JUKITO, jukitoAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.GWTP, gwtpAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.GQUERY, gqueryAnchor);
-        setAnchor(currentPlaceRequest, NameTokens.GSSS, gsssAnchor);
-    }
-
-    private void setAnchor(PlaceRequest currentPlaceRequest, String brand, AnchorElement anchor) {
-        PlaceRequest placeRequest = new PlaceRequest.Builder(currentPlaceRequest)
-                .with(NameTokens.PARAM_BRAND, brand)
-                .build();
-
-        anchor.setHref("#" + tokenFormatter.toPlaceToken(placeRequest));
-    }
+    private final AppMessages appMessages;
+    private final List<PricePresenter> products;
 
     @Inject
     HomeView(
             Binder uiBinder,
             PriceWidgetFactory priceWidgetFactory,
             PlaceManager placeManager,
-            TokenFormatter tokenFormatter) {
+            TokenFormatter tokenFormatter,
+            AppMessages appMessages) {
         this.placeManager = placeManager;
         this.tokenFormatter = tokenFormatter;
+        this.appMessages = appMessages;
+
+        products = new ArrayList<>();
 
         ProductDto shirt = new ProductDto();
-        shirt.setBrandName("Bee");
+        shirt.setBrandName(appMessages.brandName(Brand.ARCBEES));
         shirt.setPrice(55);
         shirt.setProductName("Shirt");
 
         ProductDto bag = new ProductDto();
-        bag.setBrandName("Bee");
+        bag.setBrandName(appMessages.brandName(Brand.ARCBEES));
         bag.setPrice(35);
         bag.setProductName("Bag");
 
         shirtPrice = priceWidgetFactory.create(shirt);
         bagPrice = priceWidgetFactory.create(bag);
 
+        products.add(shirtPrice);
+        products.add(bagPrice);
+
         initWidget(uiBinder.createAndBindUi(this));
     }
+
+    @Override
+    public void changeBrand(Brand brand) {
+        for (PricePresenter product : products) {
+            ProductDto value = product.getValue();
+
+            value.setBrandName(appMessages.brandName(brand));
+
+            product.setValue(value);
```

Comment:
It's already an event, have your PricePresenter listen for the BrandChanged event, and I believe you should use slots for the products